batched/dense/unit_test/Test_Batched_DenseUtils.hpp

```diff
@@ -111,6 +111,29 @@ void create_banded_triangular_matrix(InViewType& in, OutViewType& out,
   }
   Kokkos::deep_copy(out, h_out);
 }
+
+template <typename InViewType, typename OutViewType>
+void create_diagonal_matrix(InViewType& in, OutViewType& out, int k = 0) {
```

Comment:
You need to document this function a bit, what are the preconditions for `in`, `out` and what is `k`?

Reply:
I have added a docstring for this function